core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/v03/request.go

```diff
@@ -62,42 +61,34 @@ func NewClient(mercuryConfig mercury.MercuryConfigProvider, httpClient mercury.H
 }
 
 func (c *client) DoRequest(ctx context.Context, streamsLookup *mercury.StreamsLookup, pluginRetryKey string) (mercury.MercuryUpkeepState, mercury.MercuryUpkeepFailureReason, [][]byte, bool, time.Duration, error) {
-	resultLen := len(streamsLookup.Feeds)
-	ch := make(chan mercury.MercuryData, resultLen)
 	if len(streamsLookup.Feeds) == 0 {
 		return mercury.NoPipelineError, mercury.MercuryUpkeepFailureReasonInvalidRevertDataInput, [][]byte{}, false, 0 * time.Second, fmt.Errorf("invalid revert data input: feed param key %s, time param key %s, feeds %s", streamsLookup.FeedParamKey, streamsLookup.TimeParamKey, streamsLookup.Feeds)
 	}
-	resultLen = 1
+	resultLen := 1 // Only 1 multi-feed request is made for all feeds
+	ch := make(chan mercury.MercuryData, resultLen)
 	c.threadCtrl.Go(func(ctx context.Context) {
 		c.multiFeedsRequest(ctx, ch, streamsLookup)
 	})
 
 	var reqErr error
 	var retryInterval time.Duration
 	results := make([][]byte, len(streamsLookup.Feeds))
-	retryable := true
-	allSuccess := true
```

Comment:
no longer need to have logic for allSuccess since we don't have a loop and its based on single result